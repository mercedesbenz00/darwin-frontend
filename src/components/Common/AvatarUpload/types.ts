export type AvatarUploadData = {
  hash: string
  file: File
  type: string | null
}
