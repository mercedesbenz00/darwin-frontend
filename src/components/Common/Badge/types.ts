import { RGBA } from '@/utils'

export type ParsedColorType = {
  bg: string,
  text: string
}

export type BadgeType = {
  id?: string,
  label: string,
  color?: RGBA
  size?: 'small' | 'medium' | 'large'
  highContrast?: boolean
}

export enum BadgeSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
}

export enum BadgeTag {
  A = 'a',
  DIV = 'div',
  ROUTER_LINK = 'router-link',
  BUTTON = 'button'
}

export interface BadgesProps {
  items: BadgeType[],
  selected?: boolean,
  wrap?: boolean
}

export interface BadgeProps {
  label: string,
  color?: RGBA,
  size?: string,
  tag?: string,
  tooltip?: string,
  highContrast?: boolean,
  overrideLabelColor?: string,
  deletable?: boolean,
  noTooltip?: boolean,
  alpha?: number
}
