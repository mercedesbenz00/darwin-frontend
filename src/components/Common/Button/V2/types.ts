export type ButtonProps = {
  tag: ButtonTag | null,
  variant: ButtonVariant | null,
  color: ButtonColor | null,
  // style is a reserved word :(
  flair: ButtonFlair | null,
  size: CustomButtonSize | IconButtonSize | null
  fullWidth: boolean | null,
  disabled?: boolean
}

export enum ButtonVariant {
  DEFAULT = 'default',
  OUTLINE = 'outline'
}

export enum ButtonColor {
  TRANSPARENT = 'transparent',
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  POSITIVE = 'positive',
  NEGATIVE = 'negative',
  WARNING = 'warning',
  INFORMATIVE = 'informative',
  STAGE_ANNOTATION = 'stage-annotation',
  STAGE_CODE = 'stage-code',
  STAGE_COMPLETE = 'stage-complete',
  STAGE_DATASET = 'stage-dataset',
  STAGE_DISCARD = 'stage-discard',
  STAGE_LOGIC = 'stage-logic',
  STAGE_MODEL = 'stage-model',
  STAGE_NEW = 'stage-new',
  STAGE_REVIEW = 'stage-review',
}

// the icon button includes a mini
// version as it's only ment to hold
// a small icon as default slot
export enum IconButtonSize {
  MINI = 'mini',
  VERY_SMALL = 'very-small',
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
}

export enum CustomButtonSize {
  VERY_SMALL = 'very-small',
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
}

export enum ButtonFlair {
  SUPER_SOFT = 'super-soft',
  SOFT = 'soft',
  ROUNDED = 'rounded'
}

export enum ButtonTag {
  BUTTON = 'button',
  ROUTER = 'router-link',
  A = 'a',
  DIV = 'div',
}
