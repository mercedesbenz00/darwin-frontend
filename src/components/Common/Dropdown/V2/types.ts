export type DropdownOption = {
  data?: any
  id: string | number
  label: string
}

export type LegacyDropdownOption = {
  data?: any
  id: string | number
  text: string
}
