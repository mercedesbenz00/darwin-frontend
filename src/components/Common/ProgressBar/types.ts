export enum ProgressBarVariant {
  INACTIVE = 'inactive',
  ACTIVE = 'active'
}

export type ProgressBarProps = {
  value: number
  variant: ProgressBarVariant
}
