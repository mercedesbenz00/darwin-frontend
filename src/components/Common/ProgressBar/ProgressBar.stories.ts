import { Meta, Story } from '@storybook/vue'

import { ProgressBarVariant } from '@/components/Common/ProgressBar/types'

import ProgressBar from './ProgressBar.vue'

export default {
  title: 'Common/ProgressBar',
  argTypes: {
    variant: {
      control: {
        type: 'select'
      },
      options: Object.entries(ProgressBarVariant).map((val) => val[1]),
      description: 'Select the progress bar variant. Choose between active and inactive',
      table: {
        type: { summary: 'ProgressBarVariant' },
        defaultValue: { summary: 'ProgressBarVariant.ACTIVE' }
      }
    },
    value: {
      control: {
        type: 'range',
        min: 0,
        max: 1,
        step: 0.01
      },
      description: 'Value of progress bar. Decimal 0 to props.max',
      table: {
        type: { summary: 'ProgressBarVariant' },
        defaultValue: { summary: 'ProgressBarVariant.ACTIVE' }
      }
    }
  }
} as Meta<typeof ProgressBar>

export const Default: Story = (args, { argTypes }) => ({
  components: { ProgressBar },
  props: Object.keys(argTypes),
  template: "<progress-bar v-bind='$props' />"
})
Default.args = {
  value: 0.25,
  variant: ProgressBarVariant.ACTIVE
}
