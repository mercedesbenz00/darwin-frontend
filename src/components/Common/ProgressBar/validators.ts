import { ProgressBarVariant } from '@/components/Common/ProgressBar/types'

export const variantValidator = (value: ProgressBarVariant) =>
  [ProgressBarVariant.INACTIVE, ProgressBarVariant.ACTIVE].includes(value)

export const valueValidator = (value: number) => value >= 0 && value <= 1
