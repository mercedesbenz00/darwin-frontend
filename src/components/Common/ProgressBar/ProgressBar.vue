<template>
  <div
    role="progressbar"
    class="progress-indicator"
    tabindex="1"
  >
    <span
      class="progress-indicator-value"
      :class="`progress-indicator-value--${variant}`"
      :style="style"
    />
  </div>
</template>

<script lang='ts'>
import Vue from 'vue'
import { Component, Prop } from 'vue-property-decorator'

import { ProgressBarProps, ProgressBarVariant } from '@/components/Common/ProgressBar/types'
import { valueValidator, variantValidator } from '@/components/Common/ProgressBar/validators'

type ProgressBarStyle = {
  backgroundColor?: string
  width: string
}

@Component({ name: 'progress-bar' })
export default class ProgressBar extends Vue {
  @Prop({ default: 0, type: Number, validator: valueValidator })
  value!: ProgressBarProps['value']

  @Prop({ type: String, default: ProgressBarVariant.ACTIVE, validator: variantValidator })
  variant!: ProgressBarVariant

  toWhole (num: number): number {
    return num * 100.0
  }

  get style (): ProgressBarStyle {
    const percentage = this.toWhole(this.value)
    if (percentage === 100.0) {
      return {
        backgroundColor: '#25A759',
        width: '100%'
      }
    }
    return {
      width: `${percentage}%`
    }
  }
}
</script>

<style lang='scss' scoped>
@mixin base($color) {
  transition: background-color 175ms ease, width 1s ease;

  display: block;
  height: 6px;
  border-radius: 16px;
  background-color: $color;
}

.progress-indicator {
  @include base($colorSurfaceRaise);

  position: relative;
  width: 100%;
  overflow: hidden;
}

.progress-indicator-value {
  @include base($colorSurfaceRaise);

  position: absolute;
  top: 0px;
  left: 0px;

  &--inactive {
    background-color: $colorContentDisabled;
  }

  &--active {
    background-color: $colorStatusInformative;
  }
}
</style>
