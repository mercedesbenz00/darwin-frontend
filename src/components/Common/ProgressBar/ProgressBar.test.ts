import { shallowMount, Wrapper } from '@vue/test-utils'
import Vue from 'vue'

import ProgressBar from '@/components/Common/ProgressBar/ProgressBar.vue'
import { ProgressBarProps, ProgressBarVariant } from '@/components/Common/ProgressBar/types'

let wrapper: Wrapper<Vue>

const propsData: ProgressBarProps = {
  value: 0.1,
  variant: ProgressBarVariant.ACTIVE
}

beforeEach(() => {
  wrapper = shallowMount(ProgressBar, { propsData })
})

afterEach(() => {
  wrapper.destroy()
})

describe('ProgressBar UI', () => {
  it('should exist', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have correct width', async () => {
    const child = wrapper.find('.progress-indicator-value')
    expect(child.attributes().style).toContain('width: 10%;')

    await wrapper.setProps({ variant: ProgressBarVariant.ACTIVE, value: 0.5 })

    expect(child.attributes().style).toContain('width: 50%;')
  })

  it('should match class with variant', async () => {
    const child = wrapper.find('.progress-indicator-value')
    expect(child.attributes().class).toContain('progress-indicator-value--active')

    await wrapper.setProps({ variant: ProgressBarVariant.INACTIVE, value: 0.5 })

    expect(child.attributes().class).toContain('progress-indicator-value--inactive')
  })
})
