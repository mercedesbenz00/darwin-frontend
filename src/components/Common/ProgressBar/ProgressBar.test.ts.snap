// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar UI should match snapshot 1`] = `<div role="progressbar" tabindex="1" class="progress-indicator"><span class="progress-indicator-value progress-indicator-value--active" style="width: 10%;"></span></div>`;
